osu.Game/Rulesets/Edit/Checks/CheckPreviewTime.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Checks.Components;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckPreviewTime : ICheck
+    {
+        public CheckMetadata Metadata => new CheckMetadata(CheckCategory.Timing, "Check Preview Time Consistency");
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplatePreviewTimeConflict(this),
+            new IssueTemplateHasNoPreviewTime(this),
+        };
+
+        public IEnumerable<Issue> Run(BeatmapVerifierContext context)
+        {
+            var diffList = context.Beatmap.BeatmapInfo.BeatmapSet?.Beatmaps ?? new List<BeatmapInfo>();
+            int previewTime = context.Beatmap.BeatmapInfo.Metadata.PreviewTime;
+
+            if (previewTime == -1)
+            {
+                yield return new IssueTemplateHasNoPreviewTime(this).Create();
+            }
+
+            foreach (var diff in diffList)
+            {
+                if (diff.Equals(context.Beatmap.BeatmapInfo))
+                {
+                    continue;
+                }
+
+                if (diff.Metadata.PreviewTime != previewTime)
+                {
+                    yield return new IssueTemplatePreviewTimeConflict(this).Create(diff.DifficultyName);
+                }
+            }
+        }
+
+        public class IssueTemplatePreviewTimeConflict : IssueTemplate
+        {
+            public IssueTemplatePreviewTimeConflict(ICheck check)
+                : base(check, IssueType.Warning, "Audio preview time conflicts with {0} diff")
```

Comment:
I'm not sure set-level checks should live in individual difficulties' check screens. Maybe somewhere like https://github.com/ppy/osu/issues/18903 would be better.

Reply:
I am currently designed according to the behaviour of aimod. If it will affect future development, should it need to delete for now?

Reply:
I think it's okay for now. Our end goal was to move these kinds of things to set-level metadata, but until that exists having these check is likely beneficial to a mapper.

Reply:
That said, I think the message should show the differing values.

Something like

```
Audio preview time (12,345ms) doesn't match \"{diffName}\"'s preview time (23,456ms)
```